filebeat/docs/inputs/input-filestream-file-options.asciidoc

```diff
@@ -452,3 +452,60 @@ Set the location of the marker file the following way:
 ----
 file_identity.inode_marker.path: /logs/.filebeat-marker
 ----
+
+==== Log rotation
+
+As log files are constantly written, they must be rotated and purged to prevent
+the logger application from filling up the disk. Rotation is done by an external
+application, thus, {beatname_uc} needs information how to cooperate with it.
+
+When reading from rotating files make sure the paths configuration includes
+both the active file and all rotated files.
+
+By default, {beatname_uc} is able to track files correctly in the following strategies:
+* create: new active file with a unique name is created on rotation
+* rename: rotated files are renamed
+
+However, in case of copytruncate strategy, you should provide additional configuration
+to {beatname_uc}.
+
+[float]
+===== rotation.external.strategy.copytruncate
+
+If the log rotating application copies the contents of the active file and then
+truncates the original file, use these options to help {beatname_uc} to read files
+correctly.
+
+Set the option `suffix_regex` so {beatname_uc} can tell active and rotated files apart. There are
+two supported suffix types in the input: numberic and date.
+
+====== Numeric suffix
+
+If your rotated files have an incrementing index appended to the end of the filename, e.g.
+active file `apache.log` and the rotated files are named `apache.log.1`, `apache.log.2`, etc,
+use the following configuration.
+
+[source,yaml]
+---
+rotation.external.strategy.copytruncate:
+  suffix_regex: \.\d$
+  count: 7
+---
+
+`count` is the number of rotated files. {beatname_uc} needs this information to clean the
+registry when the rotated file has been purged.
+
+====== Date suffix
+
+If the rotation date is appended to the end of the filename, e.g. active file `apache.log` and the
+rotated files are named `apache.log-20210526`, `apache.log-20210527`, etc. use the following configuration:
+
+[source,yaml]
+---
+rotation.external.strategy.copytruncate:
+  suffix_regex: \-\d{6}$
+  count: 7
+  dateformat: -20060504
```

Comment:
> what is `dateformat` used for? Shouldn't the regex already find the file? Do I still need `count` in that case?

`dateformat` is used to tell Filebeat how to parse the date at the end of the file so the input can sort the rotated files.

> Given this example format, is it `{year}{month}{day}` or `{year}{day}{month}`. Depending on the logger used we might also see formats like `{day}-{month}-{year}`, even time might be included. If we want ordering, we do require a more 'intelligent' timestamp parsing, thus we can sort based on actual time.

Yes, that is why we need `dateformat`. These are the example settings for the different suffixes you had listed above:
| suffix format | `dateformat` |
| --- | --- |
| `{year}{month}{day}` | 20060102 |
| `{year}{day}{month}` | 20060201 |
| `{day}-{month}-{year}` | 02-01-2006 |
| `{year}{month}{day}{hour}{minute}` | 200601021514 | 

> This reminds me, we have a [dtfmt package](https://github.com/elastic/beats/tree/master/libbeat/common/dtfmt), but unfortunately we never added parsing support. Another use-case for timestamp parsing would be JSON. We have 2 hard coded timestamp formats for JSON parsing, but it would be nice to allow users to configure a custom pattern.

I agree, I am happy to add it in a follow-up PR.

Reply:
:+1: 

> `200601021514`

Wow... I wouldn't call this intuitive :)

> `{year}{month}{day}{hour}{minute}`

I kinda like this more human friendly readable format. `dtfmt` splits parsing and executing dataformat patterns. Maybe (as another follow up), we want to add this to `dtfmt` as well... :)